Compile error when vulkan.hpp is included

Build Environment - Android arm64-v8a ABI

When I include vulkan.hpp , I get the following compile error

vulkan.hpp:30573:29: error: no matching constructor for initialization of 'ObjectDeleter<vk::Pipeline>'
      ObjectDeleter < Pipeline > deleter( *this, allocator );
                              ^        ~~~~~~~~~~~~~~~~
vulkan.hpp:982:7: note: candidate constructor not viable: no known conversion from 'const vk::Device' to 'vk::Pipeline' for 1st argument
        ObjectDeleter(OwnerType owner = OwnerType(), Optional < const AllocationCallbacks > allocator = nullptr)

Can anyone help me how to resolve this problem.
When I replaced ObjectDeleter < Pipeline > with ObjectDeleter < Device >, the error disappears.
But is there is some other way to resolve the error

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -3189,7 +3189,7 @@ ${i}  return createResultValue( result, ${typeVariable}s, "VULKAN_HPP_NAMESPACE:
       { "command", startUpperCase(commandData.fullName) },
       { "arguments", arguments.str() },
       { "Deleter", ddit->second.pool.empty() ? "ObjectDeleter" : "PoolDeleter" },
-      { "DeleterTemplate", ddit->second.pool.empty() ? type : commandData.className + "," + ddit->second.pool },
+      { "DeleterTemplate", ddit->second.pool.empty() ? commandData.className : commandData.className + "," + ddit->second.pool },
       { "deleterArg", ddit->second.pool.empty() ? "allocator" : "allocateInfo." + startLowerCase(ddit->second.pool) },
       { "class", commandData.className },
       { "function", commandData.reducedName }

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -30570,7 +30570,7 @@ public:
     Pipeline* buffer = reinterpret_cast<Pipeline*>( reinterpret_cast<char*>( pipelines.data() ) + createInfos.size() * ( sizeof( UniquePipeline ) - sizeof( Pipeline ) ) );
     Result result = static_cast<Result>(d.vkCreateGraphicsPipelines( m_device, static_cast<VkPipelineCache>( pipelineCache ), createInfos.size() , reinterpret_cast<const VkGraphicsPipelineCreateInfo*>( createInfos.data() ), reinterpret_cast<const VkAllocationCallbacks*>( static_cast<const AllocationCallbacks*>( allocator ) ), reinterpret_cast<VkPipeline*>( buffer ) ) );
 
-    ObjectDeleter<Pipeline> deleter( *this, allocator );
+    ObjectDeleter<Device> deleter( *this, allocator );
     for ( size_t i=0 ; i<createInfos.size() ; i++ )
     {
       pipelines.push_back( UniquePipeline( buffer[i], deleter ) );
@@ -30620,7 +30620,7 @@ public:
     Pipeline* buffer = reinterpret_cast<Pipeline*>( reinterpret_cast<char*>( pipelines.data() ) + createInfos.size() * ( sizeof( UniquePipeline ) - sizeof( Pipeline ) ) );
     Result result = static_cast<Result>(d.vkCreateComputePipelines( m_device, static_cast<VkPipelineCache>( pipelineCache ), createInfos.size() , reinterpret_cast<const VkComputePipelineCreateInfo*>( createInfos.data() ), reinterpret_cast<const VkAllocationCallbacks*>( static_cast<const AllocationCallbacks*>( allocator ) ), reinterpret_cast<VkPipeline*>( buffer ) ) );
 
-    ObjectDeleter<Pipeline> deleter( *this, allocator );
+    ObjectDeleter<Device> deleter( *this, allocator );
     for ( size_t i=0 ; i<createInfos.size() ; i++ )
     {
       pipelines.push_back( UniquePipeline( buffer[i], deleter ) );
@@ -31240,7 +31240,7 @@ public:
     SwapchainKHR* buffer = reinterpret_cast<SwapchainKHR*>( reinterpret_cast<char*>( swapchainKHRs.data() ) + createInfos.size() * ( sizeof( UniqueSwapchainKHR ) - sizeof( SwapchainKHR ) ) );
     Result result = static_cast<Result>(d.vkCreateSharedSwapchainsKHR( m_device, createInfos.size() , reinterpret_cast<const VkSwapchainCreateInfoKHR*>( createInfos.data() ), reinterpret_cast<const VkAllocationCallbacks*>( static_cast<const AllocationCallbacks*>( allocator ) ), reinterpret_cast<VkSwapchainKHR*>( buffer ) ) );
 
-    ObjectDeleter<SwapchainKHR> deleter( *this, allocator );
+    ObjectDeleter<Device> deleter( *this, allocator );
     for ( size_t i=0 ; i<createInfos.size() ; i++ )
     {
       swapchainKHRs.push_back( UniqueSwapchainKHR( buffer[i], deleter ) );

